content/docs/4_quicktips/0_ide-support/quicktip.txt

```diff
@@ -0,0 +1,75 @@
+Title: Improve IDE support
+
+----
+
+Tags: integrations
+
+----
+
+Text:
+
+Many code editors and IDEs offer support for code completion suggestions or showing meta information about your variables, method calls etc. To make the most use of it, your IDE needs to know what types your variables hold.
+
+## Templates and snippets
+
+Kirby exposes some general variables like `$page`, `$site` etc. to your templates as well as anything (link: docs/guide/templates/controllers text: returned in the page controller). Your IDE however does not automatically know what these variables stand for. You can change this by adding comments specifying their types:
+
+```php "site/templates/note.php"
+/** @var Kirby\Cms\App $kirby */
+/** @var Kirby\Cms\Site $site */
+/** @var Kirby\Cms\Page $page */
+
+foreach ($page->children() as $child) {
+	echo $child->id()
+}
```

Comment:
```suggestion
<?php
/**
 * @var Kirby\Cms\App $kirby
 * @var Kirby\Cms\Site $site
 * @var Kirby\Cms\Page $page
 */

foreach ($page->children() as $child) {
	echo $child->id();
}
```

Reply:
I agree with @afbora suggestion here. I would probably also go one step further and use a separate PHP block to keep HTML and PHP separated more clearly. 

```php
<?php
/**
 * @var Kirby\Cms\App $kirby
 * @var Kirby\Cms\Site $site
 * @var Kirby\Cms\Page $page
 */
?>

<ul>
  <?php foreach ($page->children() as $child): ?>
	  <li><?= $child->title() ?></li>
  <?php endforeach ?>
</ul>
```